starttls/dialer.go

```diff
@@ -51,3 +55,19 @@ func dialWithDialer(dialer Dialer, timeout time.Duration, network, addr string,
 
 	return conn, nil
 }
+
+func wrapDialerWithProxy(dialer Dialer, connectProxy *url.URL, tlsConfig *tls.Config) (Dialer, error) {
+	dialerOpt := http_dialer.WithDialer(dialer.(*net.Dialer))
+	tlsOpt := http_dialer.WithTls(tlsConfig)
+	if connectProxy.User != nil {
+		password, ok := connectProxy.User.Password()
+		if !ok {
+			return nil, fmt.Errorf("proxy username without password not currently supported")
```

Comment:
is anything special needed for proxying using a username without password? or does anyone even use that?

Reply:
It's potentially possible (you could specify http://username@proxy/) but the upstream go-http-dialer package doesn't appear to support it (it always takes a username and password).